Resources/public/js/views/services/ez-roleserversideviewservice.js

```diff
@@ -0,0 +1,278 @@
+         * @param {EventFacade} e
+         * @param {Role} role
+         * @param {Function} end
+         */
+        _assignRoleToEachAssignee: function (e, role, end) {
+            var userService = this.get('capi').getUserService(),
+                service = this;
+
+            Y.Array.each(e.selection, function (struct) {
+                // TODO assign with limitation
+                var limitation = null,
+                    contentTypeIdentifier = struct.contentType.get('identifier');
+
+                //var limitation = { // DUMMY SECTION DATA
+                //    "_identifier": "Section",
+                //    "values": {
+                //        "ref": [{
+                //            "_href": "/api/ezp/v2/content/sections/1",
+                //            "_media-type": "application/vnd.ez.api.Section+json"
+                //        }, {
+                //            "_href": "/api/ezp/v2/content/sections/4",
+                //            "_media-type": "application/vnd.ez.api.Section+json"
+                //        }]
+                //    }
+                //};
+                //var limitation = { // DUMMY SUBTREE DATA
+                //    "_identifier": "Subtree",
+                //    "values": {
+                //        "ref": [{
+                //            "_href": "/api/ezp/v2/content/locations/1/5/14",
+                //            "_media-type": "application/vnd.ez.api.Location+json"
+                //        }]
+                //    }
+                //};
+
+                // Convert hrefs to IDs, required by API
+                if (limitation) {
+                    Y.Array.each(limitation.values.ref, function (ref, i, refs) {
+                        if (limitation._identifier == "Section") {
+                            refs[i]._href = /\d+$/.exec(ref._href)[0];
+                        } else if (limitation._identifier == "Subtree") {
+                            refs[i]._href = /(\/\d+)+$/.exec(ref._href)[0];
+                        }
+                    });
+                }
+
+                if (contentTypeIdentifier === 'user') {
+                    service._assignRoleToUser(userService.newRoleAssignInputStruct(role, limitation), struct, end);
+                } else if (contentTypeIdentifier === 'user_group') {
+                    service._assignRoleToUserGroup(userService.newRoleAssignInputStruct(role, limitation), struct, end);
+                } else {
+                    service._error('The selected content "' + struct.content.get('name') + '" is not a user or user group.');
+                }
+            });
+        },
+
+        /**
+         * Assign the role to a user
+         *
+         * @method _assignRoleToUser
+         * @protected
+         * @param {RoleAssignInputStruct} roleAssignInputStruct
+         * @param {Content} struct
+         * @param {Function} end
+         */
+        _assignRoleToUser: function (roleAssignInputStruct, struct, end) {
```

Comment:
So the name of method can be confusing (or not?).
And again - better would be to pass directly `content` instead of `struct` as `struct` is only used as `struct.content.get('contentId')`